[Improvement]: bump the version of `sqlx` used in `shuttle-shared-db`

### Describe the improvement

Current version of `sqlx` is `0.7` while `shuttle-shared-db` uses `0.6.2`.

### Duplicate declaration

- [X] I have searched the issues and this improvement has not been requested before.

Agree, I got the following error when trying to upgrade sqlx version to 0.7:
```
error[E0308]: mismatched types
   --> src/main.rs:17:36
    |
16  | async fn actix_web(
    |          --------- arguments to this function are incorrect
17  |     #[shuttle_shared_db::Postgres] pool: sqlx::PgPool,
    |                                    ^^^^ expected `Pool<sqlx::Postgres>`, found `Pool<Postgres>`
    |
    = note: `Pool<Postgres>` and `Pool<sqlx::Postgres>` have similar names, but are actually distinct types
note: `Pool<Postgres>` is defined in crate `sqlx_core`
   --> /home/ubuntu/.cargo/registry/src/index.crates.io-6f17d22bba15001f/sqlx-core-0.6.3/src/pool/mod.rs:244:1
    |
244 | pub struct Pool<DB: Database>(pub(crate) Arc<PoolInner<DB>>);
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
note: `Pool<sqlx::Postgres>` is defined in crate `sqlx_core`
   --> /home/ubuntu/.cargo/registry/src/index.crates.io-6f17d22bba15001f/sqlx-core-0.7.1/src/pool/mod.rs:267:1
    |
267 | pub struct Pool<DB: Database>(pub(crate) Arc<PoolInner<DB>>);
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    = note: perhaps two different versions of crate `sqlx_core` are being used?
note: function defined here
   --> src/main.rs:16:10
    |
16  | async fn actix_web(
    |          ^^^^^^^^^
17  |     #[shuttle_shared_db::Postgres] pool: sqlx::PgPool,
    |                                    ------------------
```

--- a/resources/aws-rds/Cargo.toml
+++ b/resources/aws-rds/Cargo.toml
@@ -11,7 +11,7 @@ async-trait = "0.1.56"
 paste = "1.0.7"
 serde = { version = "1.0.148", features = ["derive"] }
 shuttle-service = { path = "../../service", version = "0.21.0" }
-sqlx = { version = "0.6.2", features = ["runtime-tokio-native-tls"] }
+sqlx = { version = "0.7.1", features = ["runtime-tokio-native-tls"] }
 
 [features]
 postgres = ["sqlx/postgres"]

--- a/resources/shared-db/Cargo.toml
+++ b/resources/shared-db/Cargo.toml
@@ -11,8 +11,7 @@ async-trait = "0.1.56"
 mongodb = { version = "2.3.0", optional = true }
 serde = { version = "1.0.148", features = ["derive"] }
 shuttle-service = { path = "../../service", version = "0.21.0" }
-sqlx = { version = "0.6.2", optional = true }
-
+sqlx = { version = "0.7.1", optional = true }
 [features]
 postgres = ["sqlx/postgres", "sqlx/runtime-tokio-native-tls"]
 postgres-rustls = ["sqlx/postgres", "sqlx/runtime-tokio-rustls"]

--- a/resources/shared-db/Cargo.toml
+++ b/resources/shared-db/Cargo.toml
@@ -12,6 +12,7 @@ mongodb = { version = "2.3.0", optional = true }
 serde = { version = "1.0.148", features = ["derive"] }
 shuttle-service = { path = "../../service", version = "0.21.0" }
 sqlx = { version = "0.7.1", optional = true }
+
 [features]
 postgres = ["sqlx/postgres", "sqlx/runtime-tokio-native-tls"]
 postgres-rustls = ["sqlx/postgres", "sqlx/runtime-tokio-rustls"]

